Zend/zend_inheritance.c

```diff
@@ -746,9 +751,7 @@ static inheritance_status zend_do_perform_implementation_check(
 	/* Checks for constructors only if they are declared in an interface,
 	 * or explicitly marked as abstract
 	 */
-	ZEND_ASSERT(!((fe->common.fn_flags & ZEND_ACC_CTOR)
-		&& ((proto->common.scope->ce_flags & ZEND_ACC_INTERFACE) == 0
-			&& (proto->common.fn_flags & ZEND_ACC_ABSTRACT) == 0)));
+	ZEND_ASSERT(!(fe->common.fn_flags & ZEND_ACC_CTOR) || !is_abstract_or_interface_method(proto));
```

Comment:
Note to self: double-check this change. Tests didn't fail for me, but that's no guarantee.